### DISCUSSION TITLE
Is it possible to host this and access it on my browser?

### DISCUSSION THREAD

**Discussion by Heagan** [streamlink/streamlink-twitch-gui#1018] [Q&A] [answered]
My work laptop blocks twitch,

I was wondering since this was built with node could it just run as a web app and I can host it on my server?

(I wouldn't just use this to get around my work laptop, this app is also just really nice)

**Comment by Heagan** [comment]
found out vlc has a web view, `vlc -I http` and even got vanilla streamlink to open it using those arguments, 
vlc was complaining my server doesn't have an audio device,

**Comment by bastimeyer** [comment, marked as answer]
I suggest you take a look at the project's readme and read again what this application actually is and what it does. None of your questions make any sense. This is the **Streamlink Twitch GUI** project, which is an NW.js (Chromium+NodeJS) based web-browser for Twitch.tv on Linux/macOS/Windows, and a launcher for the Python-based **Streamlink** application. Streamlink Twitch GUI has nothing to do with watching streams itself. This is done by Streamlink.

If you want to bypass network restrictions on your local machine, then just use a proxy/VPN. Simple as that. If you already own a server, then this should be trivial.

When [using OpenSSH, you can locally create a socks5 tunnel to your server](https://man.archlinux.org/man/core/openssh/ssh.1.en#D) and [use that via Streamlink](https://streamlink.github.io/cli/proxy.html) like this

```sh
$ ssh -D $PORTNUMBER -NTCqf "$USER@$SSHSERVER"
$ streamlink --http-proxy "socks5h://localhost:$PORTNUMBER" twitch.tv/... best
```

as well as in Streamlink Twitch GUI via the `--proxy-server` launch argument, so you can browse streams using the socks5 tunnel

```sh
$ streamlink-twitch-gui "--proxy-server=socks5://localhost:$PORTNUMBER"
```

Set the `--http-proxy` Streamlink launch argument in the Twitch GUI's streaming settings menu after enabling advanced settings in the main settings menu first, or add it to your [Streamlink config file](https://streamlink.github.io/cli/config.html), so it'll be picked up automatically.

Good luck.

**Comment by Heagan** [comment]
Ah thank you, sorry I'm not the smartest, so my understanding probably isn't very good, 

I thought since its built with HTML, JS, CSS as well as Chromium there may be a way to just have it so I could access the GUI on my browser

The laptop I'm using is super restrictive (SSH disabled) and its forced to already use a specific VPN, 

I can access my server through the browser so I thought maybe I could get streamlink gui working

**Reply by Heagan** [reply]
streaklink on my server works, it can download streams, just needed a way to stream it through my browser 😅

**Reply by bastimeyer** [reply]
There's the [`--player-external-http`](https://streamlink.github.io/cli.html#cmdoption-player-external-http) launch argument, but you won't be able to watch it via your web browser because of the stream container format.

You can use any local video player with support for HTTP streaming though, like VLC or MPV. Just make sure that the HTTP server spawned by the Streamlink process is not accessible to anyone else. Only watching one stream at a time is supported. That's because the option is merely an extension for the `--player-http` launch argument, except for allowing to watch on a different machine on the network.

**Reply by Heagan** [reply]
it worked! thank you, the video is very stuttery, only showing 1 second before freezing again, but I think thats my VLC settings so I'll try a few things, but this solution works
